crates/graphql-composition/src/emit_federated_graph/federation_builtins.rs

```diff
@@ -0,0 +1,257 @@
+use super::*;
+use graphql_federated_graph::Wrapping;
+
+pub(super) fn emit_federation_builtins(ctx: &mut Context<'_>) {
+    let string_definition = ctx.definitions[&ctx.lookup_str("String").expect("String to be defined")];
+    let boolean_definition = ctx.definitions[&ctx.lookup_str("Boolean").expect("Boolean to be defined")];
+    let graph_str = ctx.insert_str("graph");
+    let name_str = ctx.insert_str("name");
+    let url_str = ctx.insert_str("url");
+
+    // join__FieldSet
+    let join_fieldset_scalar = {
+        let name = ctx.insert_str("join__FieldSet");
+        ctx.out.push_scalar_definition(federated::ScalarDefinitionRecord {
+            name,
+            directives: Vec::new(),
+            description: None,
+        })
+    };
+
+    // join__Graph
+    let join_graph_scalar = {
+        let name = ctx.insert_str("join__Graph");
+        ctx.out.push_scalar_definition(federated::ScalarDefinitionRecord {
+            name,
+            directives: Vec::new(),
+            description: None,
+        })
+    };
+
+    // directive @join__unionMember(graph: join__Graph!, member: String!) repeatable on UNION
+    {
```

Comment:
Ideally we'll do what we do in this function with SDL instead of verbose constructors when `@link` support is implemented.